src/mempool/mempool.rs

```diff
@@ -110,21 +145,33 @@ impl Mempool {
     }
 
     pub async fn run(&mut self) {
-        while let Some(message) = self.mempool_rx.recv().await {
-            if self.message_is_valid(&message) {
-                let fid = message.fid();
-                let shard = self.message_router.route_message(fid, self.num_shards);
-                let senders = self.shard_senders.get(&shard);
-                match senders {
-                    None => {
-                        error!("Unable to find shard to send message to")
-                    }
-                    Some(senders) => {
-                        if let Err(err) = senders.messages_tx.send(message).await {
-                            error!("Unable to send message to engine: {}", err.to_string())
+        loop {
+            tokio::select! {
+                message = self.mempool_rx.recv() => {
```

Comment:
Selecting directly here is dangerous, the mempool rx channel and overload the messages_request_rx channel, and make it so the request is never fulfilled if we get sufficient message traffic to the mempool.

It's better to loop using a tick and ensure that the messages request channel is processed first and always prioritized. 

See https://github.com/farcasterxyz/snapchain/blob/41f1e05950a216d57fac04c56d27f3830fea0fa5/src/consensus/proposer.rs#L284-L288 for an example of how we're done this elsewhere.

Reply:
I made a change to specify `biased` in the select macro. This causes the macro to preference branches in order of specification. I think this is what we want here? 

In this case, I don't really think we need a timeout because we will just poll indefinitely.